go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/AzureAD/microsoft-authentication-library-for-go
 
-go 1.17
+go 1.18
```

Comment:
Are we using something new in Go 1.18?

Reply:
I missed that generics were introduced in this change, so that part makes sense.

Regarding moving forward to the supported N-1 release when no version-specific features are used, I'm not quite convinced.  While I don't think we should go out of our way to support unsupported versions of Go, I don't think we should just drop them.